monkey/monkey_island/cc/setup/mongo/mongo_setup.py

```diff
@@ -18,23 +21,41 @@
 logger = logging.getLogger(__name__)
 
 
-def setup_mongodb(config_options: IslandConfigOptions):
-    if config_options.start_mongodb:
-        MongoDbRunner(
-            db_dir_parent_path=config_options.data_dir, logging_dir_path=config_options.data_dir
-        ).launch_mongodb()
-    wait_for_mongo_db_server(MONGO_URL)
-    assert_mongo_db_version(MONGO_URL)
+def start_mongodb(data_dir: str) -> MongoDbProcess:
+    db_dir = _create_db_dir(data_dir)
+    log_file = os.path.join(data_dir, MONGO_LOG_FILENAME)
+
+    mongo_db_process = MongoDbProcess(db_dir=db_dir, log_file=log_file)
+    mongo_db_process.start()
+
+    return mongo_db_process
+
+
+def _create_db_dir(db_dir_parent_path) -> str:
+    db_dir = os.path.join(db_dir_parent_path, DB_DIR_NAME)
+    logger.info(f"Database content directory: {db_dir}.")
+
+    create_secure_directory(db_dir, create_parent_dirs=False)
+    return db_dir
+
+
+def register_mongo_shutdown_callback(mongo_db_process: MongoDbProcess):
+    atexit.register(mongo_db_process.stop)
```

Comment:
Nice, didn't know python had such functionality. What does this achieve in practice (gracefully killing the process I mean)? Also, maybe this should be in `mongo_setup.py`, since it's on process level of abstraction and is not valid if `mongo_db_process.start` wasn't called.

Reply:
> What does this achieve in practice (gracefully killing the process I mean)?

We're ensuring that Mongo is actually shut down when Monkey Island is shut down. In most cases, the OS will handle this for you, but it's a best practice not to leave processes laying round when you're done with them.

> Also, maybe this should be in `mongo_setup.py`

I think there's a typo somewhere in your comment. It is in `mongo_setup.py`.

Reply:
@VakarisZ and I discussed moving `register_mongo_shutdown_callback()` into the `MongoDbProcess` class. We decided that the responsibility of `MongoDbProcess` is to provide an interface that allows the MongoDB process to be started and stopped. It is not the responsibility of `MongoDbProcess` to manage the process. The responsibility of managing different processes and threads within Monkey Island is better left to a component at a higher level of the architecture.